packages/fxa-auth-server/lib/payments/stripe.ts

```diff
@@ -1582,66 +1528,11 @@ export class StripeHelper {
     uid: string,
     expand?: ('subscriptions' | 'invoice_settings.default_payment_method')[]
   ): Promise<Stripe.Customer | void> {
-    const { stripeCustomerId } = (await getAccountCustomerByUid(uid)) || {};
-    if (!stripeCustomerId) {
-      return;
-    }
-
-    // By default this has subscriptions expanded.
-    let customer = await this.expandResource<Stripe.Customer>(
-      stripeCustomerId,
-      CUSTOMER_RESOURCE
-    );
-
-    if (customer.deleted) {
-      await deleteAccountCustomer(uid);
-      return;
-    }
-
-    // If the customer has subscriptions and no currency, we must have a stale
-    // customer record. Let's update it.
-    if (customer.subscriptions?.data.length && !customer.currency) {
-      await this.stripeFirestore.fetchAndInsertCustomer(customer.id);
-      // Retrieve the customer again.
-      customer = await this.expandResource<Stripe.Customer>(
-        stripeCustomerId,
-        CUSTOMER_RESOURCE
-      );
-    }
-
-    // Since the uid is just metadata and it isn't required when creating a new
-    // customer _on Stripe dashboard_, we have an edge case where the customer
-    // is created on Stripe and locally via the `customer.created` event, but
-    // the uid metadata is still missing.  Throwing an error here causes a
-    // profile fetch to fail, thus would block the user completely.
-    //
-    // Customers created through our regular flow will always have their uid in
-    // the metadata.
-    //
-    // So, we'll only throw an error if the uid metadata is found and it does
-    // not match.
-
-    if (customer.metadata.userid && customer.metadata.userid !== uid) {
-      // Duplicate email with non-match uid
-      const err = new Error(
-        `Stripe Customer: ${customer.id} has mismatched uid in metadata.`
-      );
+    try {
+      return await super.fetchCustomer(uid, expand);
```

Comment:
This isn't `this.fetchCustomer`? There's two implementations?

Reply:
It's hijacking the base's function. If I were to call this.fetchCustomer it would just be a never ending recursive call and would blow up the call stack.